src/dbnode/encoding/iterators.go

```diff
@@ -224,6 +230,10 @@ func (i *iterators) validateNext(next bool, prevAt xtime.UnixNano) (bool, error)
 	return next, nil
 }
 
+func (i *iterators) firstAnnotation() ts.Annotation {
+	return i.firstAnnotationHolder.get()
```

Comment:
I think it would be good to panic in case annotation is being accused before `Next` was invoked. Otherwise we might be hiding bug.